parlai/core/agents.py

```diff
@@ -325,6 +325,9 @@ def create_agent_from_opt_file(opt: Opt):
             opt_from_file[k] = v
 
     opt_from_file['model_file'] = model_file  # update model file path
+    if opt.get('init_model') is not None:
```

Comment:
a couple quick questions:

1) why is this not handled in lines 309-315?
2) why is this not handled in lines 323-325?

Reply:
1. when train_model.py sets opt['init_model'] to the checkpoint, it does not add it into override. adding to override could be one possible solution, what do you think?
2. it isn't handled here in the specific case where init_model *did* exist in opt_from_file (i.e. (1) the PR description). ex: finetuning a model on BST that's initialized with Reddit and it gets requeued.

Reply:
I dont know which is better, but i'm very slightly leaning towards adding it to `override` in train model, it feels marginally better than carving out a special case for `init_model` in these `opt` functions. do you have an opinion either way?

Reply:
I don't really know if I like either. I think adding to override is hacky and can't be trusted lol. But also carving out this special exception isn't very nice either... @stephenroller ?

Reply:
Both seem equally gross to me. This solution, with comments explaining wtf is going on, and a test checking that we do it correctly, seems sufficient.

Should we kill override, then we'll at least catch doing it wrong in the future.